### PR TITLE
Fix sections not showing up on instructor search

### LINKED ISSUE #9: BUG: Search by Instructor's '+' button in table does nothing
### Summary

While performing any search, courses in the results table that contain sections will be distinguished by a '+' button on the left side of the row. Clicking the button will reveal the section times for that course.
If you perform a Search by Instructor, the same + buttons will appear for courses that have section times. Clicking this drop down results in nothing happening. 
My guess is that the reason nothing happens is while performing a search by instructor, only courses and sections with the searched for name show up in the table. So even though a course appears with a + button, no sections appear because the section TA's almost always have a different name from the professor.

### Expected Behavior

There are two potential solutions to this.

1. Remove the + sign from all "Search by Instructor" searches.

2. Have sections show up in the drop down despite not having the same name as the instructor search.

### Current / Observed Behavior

If you perform a Search by Instructor, the same + buttons will appear for courses that have section times. Clicking this drop down results in nothing happening. 

### Steps to Reproduce

1. Go to Search by Instructor
2. Set start quarter to W20 and end quarter to S23
3. Set instructor search to "Conrad"
4. Click drop down buttons for the resulting rows.

See here that the first two drop downs result in nothing, while the third drop down works successfully.
image

### Browser / Platform Info

_No response_

### PR DESCRIPTION
**Pull request by rdoumeth**
Fixed search by instructor functionality by removing the expandable "+" sign, as proposed by issue #9. This was done by creating another table without the grouping. "+" sign is present in other searches, such as building search.

Dokku deployment:
https://proj-courses-rdoumeth.dokku-09.cs.ucsb.edu/courseovertime/instructorsearch

image

Closes #9

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
It still doesn't seem to be working properly.

Look up CMPSC 156 for W22. There were two TAs, LU and HEFFERNAN

If you search for either of them, only the 5pm section shows up.  All three should show up.

As another example, consider CMPSC 5A.  There are two sections:
image


But search for TANNA and only one shows up:

image


So the bug still isn't fixed; just replaced with a different bug.

File: frontend/src/main/components/InstructorSearchTableBase.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// Stryker disable StringLiteral, ArrayDeclaration
+export default function InstructorSearchTableBase({
+  columns,
+  data,
+  testid = "testid",
+}) {
+  // Stryker disable next-line ObjectLiteral
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    useTable(
+      {
```
[the remark this answers is not in the document]
Comment (rdoumeth) [reply frontend/src/main/components/InstructorSearchTableBase.js:14]: I actually based my code on SectionsTableBase which doesn't have that column hidden, for some reason. I'll do that change right now.

File: frontend/src/main/components/InstructorSearchTableBase.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import React, { Fragment } from "react";
+import { useTable, useExpanded } from "react-table";
+import { Table } from "react-bootstrap";
+
+// Stryker disable StringLiteral, ArrayDeclaration
```
Comment (pconrad) [inline frontend/src/main/components/InstructorSearchTableBase.js:5]: Please, no, ... this is too broad.

Use `// Stryker disable all` plus `// Stryker restore all` around the smallest section of code possible.

File: frontend/src/main/components/InstructorSearchTableBase.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          prepareRow(row);
+          return (
+            <Fragment key={`row-${i}`}>
+              <tr {...row.getRowProps()}>
+                {row.cells.map((cell, _index) => {
+                  return (
+                    <td
+                      {...cell.getCellProps()}
+                      data-testid={`${testid}-cell-row-${cell.row.index}-col-${cell.column.id}`}
+                      // Stryker disable next-line ObjectLiteral
```
Comment (pconrad) [inline frontend/src/main/components/InstructorSearchTableBase.js:52]: There is no need for a stryker exception here.  You can write a test for this quite easily; there are many examples in our code bases.  You want to look for something like `expect(thing).toHaveAttribute('style'...`

**Review by pconrad — CHANGES_REQUESTED** [review]
The code finally works!

Now that it does, there are a few things in the code that need to be fixed.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**